### ISSUE TITLE
Improve documentation about options

### ISSUE THREAD

**Issue by Vadorequest** [TypeStrong/grunt-ts#229]
It's not clear how the options should be set.

It the doc, there is column `where to define` but it's confusing (I just saw it after an hour reading that doc).

Wouldn't it be better to have two tables, one with the "options" to define in the `target` and one in the `options` ? Btw, the column `where to define` should have `target` and `options`, with a `s`, not just `option`, it is also confusing.

**Comment by nycdotnet** [comment]
We did a major rewrite of the documentation a few months ago and it's a lot better than it was, but I agree that this could still be improved.

What would you think of changing that table to something like this?  Is that better?  I already link to the example, but it's probably not obvious.

|property|description|
|:----|:-----|
|[comments](#comments)|`true`, `false` (default) - include comments in emitted JS.  Set in `options`; [see example](#commentsexample)|
|[files](#files)|Sets of files to compile and optional output destination. Set on target; [see example](#filesexample)|

**Comment by Vadorequest** [comment]
I tink the current way is fine, just split it into two differents tables, just so it's clear what should be where. Also add a full example with all options set would be great, I think.

**Comment by basarat** [comment]
> Also add a full example with all options set would be great,

There are a lot of options not all which can be set together (e.g out and module are really exclusive)

**Comment by Vadorequest** [comment]
True, but still, an example with all settings with default values would
work :)​
All options can be set at the same time even though they can't be enabled
at the same time.

And it would make it much easier, just copy/paste the whole stuff and
change what we want to custom.

**Comment by nycdotnet** [comment]
> True, but still, an example with all settings with default values would
> work :)​

We could probably do this, but I'm not convinced it would actually be better.  I think that pushing our current main Gruntfile and the sample one might better serve the users.
